src/formElementPlugins/ApplicationViewWrapperNEW.tsx

```diff
@@ -149,15 +149,72 @@ const ApplicationViewWrapper: React.FC<ApplicationViewWrapperPropsNEW> = (props)
     />
   )
 
+  const changesToResponseProps: ChangesToResponseWarningProps = {
+    currentReview,
+    isChanged,
+    isValid: validationState ? (validationState.isValid as boolean) : true,
+  }
+
+  const displayResponseWarning = !!currentReview || (isChanged && validationState.isValid)
+
   return (
     <ErrorBoundary pluginCode={pluginCode}>
       <React.Suspense fallback="Loading Plugin">
-        {parametersReady && <Form.Field required={isRequired}>{PluginComponent}</Form.Field>}
+        <div
+          style={{
+            border: displayResponseWarning ? 'solid 1px' : 'transparent',
+            borderColor: isChanged ? 'blue' : !!currentReview ? 'red' : 'black',
+            padding: '5px',
+            margin: '5px',
+          }}
+        >
+          {parametersReady && <Form.Field required={isRequired}>{PluginComponent}</Form.Field>}
+          <ChangesToResponseWarning {...changesToResponseProps} />
+        </div>
       </React.Suspense>
     </ErrorBoundary>
   )
 }
 
+interface ChangesToResponseWarningProps {
+  currentReview?: ReviewResponse
+  isChanged: boolean
+  isValid: boolean
+}
+
+const ChangesToResponseWarning: React.FC<ChangesToResponseWarningProps> = ({
+  currentReview,
+  isChanged,
+  isValid,
+}) => {
+  const displayResponseWarning = currentReview || (isChanged && isValid)
+
+  return (
+    <div
+      style={{
+        color: isChanged ? 'grey' : 'red',
+        minHeight: 18, // Used to keep space for warning when not displaying
```

Comment:
Setting a minHeight is probably not the best way to do this, as there's no way to guarantee pixel heights of text and icons between browsers. For example, in Firefox I still see a slight jump.
I'm guessing you used `display:none` and it didn't achieve what you wanted? I think I told you wrong -- you should use `visibility: hidden` instead. That keeps the space but just hides the element, whereas `display:none` removes it from rendering completely.

Reply:
> Setting a minHeight is probably not the best way to do this

Oops. I wasn't sure what was the best approach. 

> I'm guessing you used display:none and it didn't achieve what you wanted?

I didn't : ( That's exact why I did that. But it's not looking good indeed. After I added the border it started to jump in Chrome too.

> you should use visibility: hidden instead. That keeps the space but just hides the element, whereas display:none removes it from rendering completely.

Will try that. Thanks!

Reply:
It works well. : ) Done: 7c5859b

Reply:
Yeah great. Tiny picky thing -- there's still the smallest jump due to the 1px border adding to the padding. Can fix with:
`padding: displayResponseWarning ? 4 : 5,`
😉

Reply:
Cool doing that now : )